aws/aws-explanation/canonical-offerings.rst

```diff
@@ -1,71 +1,82 @@
 
-**For EKS** - These are optimised AMIs for running Kubernetes. They uses a slimmed-down minimal image, a custom kernel and are optimised for performance and security on EKS clusters.
+`Ubuntu Pro images`_
+--------------------
 
-`Ubuntu Pro images`_ - These are premium images that include certified components, hardening options, comprehensive open source security coverage (for packages in main and universe) for 10 years, kernel Livepatch service and optional 24/7 enterprise-grade support. They are available on AMD64 and Graviton architectures, both with an optimised kernel to provide better performance while supporting nearly all instance types available
+These are premium images that include certified components, hardening options, comprehensive open source security coverage (for packages in main and universe) for 10 years, `kernel Livepatch service`_ and optional 24/7 enterprise-grade support. They are available on amd64 and arm64 (Graviton) architectures, both with an optimised kernel to provide better performance while supporting nearly all instance types available. Those premium images are available via the `AWS Marketplace`_ and
+also via the `EC2 console <https://ubuntu.com/blog/ubuntu-pro-is-now-part-of-the-aws-ec2-console>`_.
 
-`Ubuntu Pro FIPS images`_ - These are FIPS certified images used to meet compliance requirements in production environments. These are available only through the AWS Marketplace or through private offers.
+`Ubuntu Pro FIPS images`_
+-------------------------
 
-Optimised kernel
-----------------
+These are FIPS certified images used to meet compliance requirements in production environments. These are available only through the `AWS Marketplace`_ or through private offers.
 
-The ``linux-aws`` kernel enables the following AWS specific features:
+Ubuntu Pro with Real-time Kernel
+--------------------------------
 
-`Elastic Fabric Adapter`_ - This allows high-performance applications to directly access the network adapter and get reliable transport with low-latency. 
+Theses are images with an `real-time kernel <https://ubuntu.com/real-time>`_. Those images are available
+via the `AWS Marketplace`_.
 
-`Nitro enclaves driver`_ - This gives data processing applications a secure enclave with CPU and memory isolation to prevent data leaks.
+EKS images
+----------
 
-`AWS Graviton`_ - The arm64 version of the kernel includes several patches to take advantage of the unique features of AWS Graviton native CPUs.
+These are optimised AMIs for running as worker nodes with the Amazon Elastic Kubernetes Service (EKS).
+They uses a slimmed-down minimal image, the AWS-optimized kernel and are optimised for performance
+and security on EKS clusters. Those images do contain the required software (eg `kubelet` and `kubectl`)
+to run as a Kubernetes worker node. The images are availalbe via the `AWS Marketplace`_ or the `EC2 console <https://cloud-images.ubuntu.com/docs/aws/eks/>`_.
```

Comment:
This `EC2 console` link seems wrong. Also, this is probably the reason for the warning, as `EC2 console` was used earlier on line 15 too, but it was pointing to a different link (blog).
It is preferable to define all links at the bottom, to keep the text readable and for easy maintenance. It'll hopefully help avoid such warnings too.

Reply:
If I add the link to the bottom, how can I use the same text to link to different targets?

Reply:
Oh! Usually the text should try to represent the contents of the target (while also creating a meaningful statement that we are trying to convey), so that the user can decide whether they want to click on it or not. So ideally we should try and avoid the same text for different targets. 

But you are right, if you do want to use the same text for different targets, then adding the link to the bottom won't work. But then the warning will still be generated. So we'll either have to (i) remove that warning check or (ii) change the RTD build to not fail on warnings - both of which I am not too keen on doing because these are useful checks.

Reply:
found a solution. there are anonymous links (ending with double underscore `__`). Does that work for you?